site/src/pages/RoadmapPage/SearchSidebar.tsx

```diff
@@ -1,65 +1,95 @@
 import './SearchSidebar.scss';
 
-import CloseButton from 'react-bootstrap/CloseButton';
-import SearchHitContainer from '../../component/SearchHitContainer/SearchHitContainer';
 import SearchModule from '../../component/SearchModule/SearchModule';
-import CourseHitItem from './CourseHitItem';
 
 import { useIsMobile } from '../../helpers/util';
 import { useAppDispatch, useAppSelector } from '../../store/hooks';
-import { setShowSearch } from '../../store/slices/roadmapSlice';
-import { StrictModeDroppable } from './StrictModeDroppable';
-import CourseBag from './CourseBag';
+import { setActiveCourse, setShowSearch } from '../../store/slices/roadmapSlice';
+import { useEffect, useRef } from 'react';
+import UIOverlay from '../../component/UIOverlay/UIOverlay';
 
-const SearchSidebar = () => {
+import { ReactSortable, SortableEvent } from 'react-sortablejs';
+import { useCoursebag } from '../../hooks/coursebag';
+import { CourseGQLData } from '../../types/types';
+import Course from './Course';
+import { courseSearchSortable } from '../../helpers/sortable';
+import { Spinner } from 'react-bootstrap';
+import { useNamedAcademicTerm } from '../../hooks/namedAcademicTerm';
+
+const CloseRoadmapSearchButton = () => {
+  const isMobile = useIsMobile();
   const dispatch = useAppDispatch();
+  const { year, quarter } = useNamedAcademicTerm();
+
+  if (!isMobile) return <></>;
+
+  const closeSearch = () => dispatch(setShowSearch({ show: false }));
+
+  return (
+    <button className="fixed" onClick={closeSearch}>
+      Cancel Selecting for {quarter} {year}
+    </button>
+  );
+};
+
+const SearchSidebar = () => {
   const isMobile = useIsMobile();
+  const showSearch = useAppSelector((state) => state.roadmap.showSearch);
   const { showCourseBag } = useAppSelector((state) => state.roadmap);
+  const overlayRef = useRef<HTMLDivElement>(null);
+  const sidebarRef = useRef<HTMLDivElement>(null);
+  const dispatch = useAppDispatch();
+
+  const { coursebag } = useCoursebag();
+  const { results, searchInProgress } = useAppSelector((state) => state.search.courses);
+  const shownCourses = JSON.parse(JSON.stringify(showCourseBag ? coursebag : results)) as CourseGQLData[];
```

Comment:
I assume `JSON.parse(JSON.stringify(...)` is a way to make a deep copy of the array of courses. Is that necessary? If so, add a comment stating why. Also not sure if this is the best way to make a deep copy but can't think of any other off the top of my head.